src/main/java/us/daconta/xlmeco/provider/impl/GrokProvider.java

```diff
@@ -0,0 +1,190 @@
+package us.daconta.xlmeco.provider.impl;
+
+import okhttp3.*;
+import us.daconta.xlmeco.grpc.ChatRequest;
+import us.daconta.xlmeco.grpc.ChatResponsePart;
+import us.daconta.xlmeco.grpc.ModelParameters;
+import us.daconta.xlmeco.provider.ChatProvider;
+import io.grpc.stub.StreamObserver;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import us.daconta.xlmeco.provider.EmbeddingProvider;
+import us.daconta.xlmeco.provider.GenerativeProvider;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+public class GrokProvider extends AbstractGenerativeProvider implements ChatProvider, EmbeddingProvider {
```

Comment:
Why does it implement EmbeddingProvider if it cannot provide embeddings?  Recommend removing that and setting the supports "embedding" property to false.